src/ilx/EraseUnions.fs

```diff
@@ -71,7 +69,6 @@ type UnionReprDecisions<'Union,'Alt,'Type>
 #if FLATTEN_SINGLE_NON_NULLARY_AND_ALWAYS_USE_TAGS
                 nullPermitted cu then
 #else
-                alts.Length < TaggingThresholdFixedConstant &&
```

Comment:
This change would mean that types with more than 4 cases would use type tests rather than integer tag dispatch.  That's not ok, as a type with 100 cases will now use 100 type tests to discriminate on a value, rather than a switch on a single integer.

I believe the right fix is one of these:

(1) ensure that UseNullAsTrueValue works when IntegerTag discrimination is used (by adding extra null checks when discriminating), OR

(2) always use RuntimeTypes discrimination when UseNullAsTrueValue is specified (likely easier to implement, but should be included in the docs for UseNullAsTrueValue since adding the attribute to a type with many cases - at least one of which carries data - may change performance characteristics very substantially)

BTW the use of "4" as the point where we switch discrimination technique was determined based on perf experiments on x86 back in 2005.  It may be that it's not the right point to swtich nowadays on x64.  It would be worth checking.  But you have to switch somewhere.